libraries/AP_Camera/AP_Camera_Tracking.cpp

```diff
@@ -0,0 +1,70 @@
+#include "AP_Camera_Tracking.h"
+
+#if AP_CAMERA_TRACKING_ENABLED
+
+extern const AP_HAL::HAL& hal;
+
+// set tracking to none, point or rectangle (see TrackingType enum)
+// if POINT only p1 is used, if RECTANGLE then p1 is top-left, p2 is bottom-right
+// p1,p2 are in range 0 to 1.  0 is left or top, 1 is right or bottom
+bool AP_Camera_Tracking::set_tracking(TrackingType tracking_type, const Vector2f& p1, const Vector2f& p2, uint8_t tracking_device_sysid, uint8_t tracking_device_compid, mavlink_camera_information_t _cam_info)
+{
+    // if we don't support the required tracking then return
+    switch (tracking_type) {
+        case TrackingType::TRK_NONE:
+            break;
+        case TrackingType::TRK_POINT:
+            if (!(_cam_info.flags & CAMERA_CAP_FLAGS_HAS_TRACKING_POINT)) {
+               return false;
+            }
+            break;
+        case TrackingType::TRK_RECTANGLE:
+            if (!(_cam_info.flags & CAMERA_CAP_FLAGS_HAS_TRACKING_RECTANGLE)) {
+               return false;
+            }
+            break;
+        default:
+            return false;
+    }
+
+    gcs().send_text(MAV_SEVERITY_INFO,"Tracking: New Tracking request");
+    uint8_t proxy_device_compid=0, proxy_device_sysid=0;
+    auto _link = GCS_MAVLINK::find_by_mavtype_and_compid(MAV_TYPE_ONBOARD_CONTROLLER, proxy_device_compid, proxy_device_sysid);
```

Comment:
I think we shouldn't look this up each time.  Let's store this as a variable/member